src/app/archive/upgrade_to_mesa.sql

```diff
@@ -162,17 +162,39 @@ SELECT add_zkapp_states_nullable_element(30);
 SELECT add_zkapp_states_nullable_element(31);
 
 -- 3. `zkapp_states`: Add columns element8..element31
+
+CREATE OR REPLACE FUNCTION get_zero_field_id() RETURNS int AS $$
+DECLARE
+  result int;
+  zero text := '0';
+BEGIN
+  -- Try to fetch existing id
+  SELECT id INTO result FROM zkapp_field WHERE field = zero;
+
+  -- If not found, insert and get the new id
+  IF result IS NULL THEN
```

Comment:
Actually this would cause archive db to be inconsistent on ids if we upgrade first then try to fetch precomputed blocks etc. 

Problem is do we need everything in Archive DB to be 100% stable?